src/test/java/space/obminyashka/items_exchange/controller/EmailControllerTest.java

```diff
@@ -0,0 +1,74 @@
+package space.obminyashka.items_exchange.controller;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.*;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.context.MessageSource;
+import space.obminyashka.items_exchange.exception.EmailValidationCodeExpiredException;
+import space.obminyashka.items_exchange.exception.EmailValidationCodeNotFoundException;
+import space.obminyashka.items_exchange.service.MailService;
+import space.obminyashka.items_exchange.util.MessageSourceUtil;
+
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class EmailControllerTest {
+    @Mock
+    private MailService mailService;
+    @Mock
+    private MessageSource messageSource;
+    @InjectMocks
+    private MessageSourceUtil messageSourceUtil;
```

Comment:
That is not the right way to create a unit test. The test class has to be only the single one.

You should mock that source and validate that it was invoked with the correct message key.